File: string/delete-columns-to-make-sorted/delete-columns-to-make-sorted.go
package deletecolumnstomakesorted

func minDeletionSize(strs []string) int {
	nOfCols := len(strs[0])
	del := 0
	for col := 0; col < nOfCols; col++ {
		for row := 0; row < len(strs)-1; row++ {
			if strs[row][col] > strs[row+1][col] {
				del++
				break
			}
		}
	}
	return del
}

File: string/delete-columns-to-make-sorted/delete-columns-to-make-sorted.ts
function minDeletionSize(strs: string[]): number {
    const nOfCols = strs[0].length;
    let del = 0;

    for (let col = 0; col < nOfCols; col++) {
        for (let row = 0; row < strs.length - 1; row++) {
            if (strs[row][col].charCodeAt(0) > strs[row + 1][col].charCodeAt(0)) {
                del++;
                break;
            }
        }
    }

    return del;
};
